Add Default Callback Name editor setting

dalexeev opened the pull request:
Adds the "interface/editors/default_signal_callback_name" editor setting,
which allows you to specify the format of the default callback name
in the Signal Connection Dialog.

---

Available substitutions: `{NodeName}`, `{nodeName}`, `{node_name}`, `{SignalName}`, `{signalName}`, `{signal_name}`.

![](https://user-images.githubusercontent.com/47700418/134618644-f0cb28f4-19f5-4aa3-94d9-2bb226b75584.png)

Closes godotengine/godot-proposals#1023
Closes godotengine/godot-proposals#3346.

groud commented:
Is there a proposal attached to this PR ? I am not sure such customization is very useful.

dalexeev commented:
See godotengine/godot-proposals#3346.

KoBeWi commented on editor/connections_dialog.cpp:
```diff
@@ -712,38 +712,56 @@ bool ConnectionsDock::_is_item_signal(TreeItem &item) {
+	subst["signalName"] = s;
+	subst["signal_name"] = signal_name.capitalize().replace(" ", "_").to_lower();
+
+	String dst_method = String(EDITOR_DEF("interface/editors/default_callback_name", "_on_{node_name}_{signal_name}")).format(subst);
```
This should be `EDITOR_GET`. Defining something twice is a bad idea.

KoBeWi commented on editor/connections_dialog.cpp:
```diff
@@ -712,38 +712,56 @@ bool ConnectionsDock::_is_item_signal(TreeItem &item) {
+
+	String s = node_name.capitalize().replace(" ", "");
+	subst["NodeName"] = s;
+	if (s.length()) {
```
```suggestion
	if (!s.is_empty()) {
```

KoBeWi commented on editor/connections_dialog.cpp:
```diff
@@ -712,38 +712,56 @@ bool ConnectionsDock::_is_item_signal(TreeItem &item) {
+
+	s = signal_name.capitalize().replace(" ", "");
+	subst["SignalName"] = s;
+	if (s.length()) {
```
```suggestion
	if (!s.is_empty()) {
```

KoBeWi reviewed (approved):
Looks good, my only concern (aside from the comments I left) is that it's possible to put an invalid identifier as a default callback. Not sure how to safeguard against that though, other than extending the validation from `node_name` to the whole callback. Also the setting name should be `default_signal_callback_name`, to be more clear.

Though I'd wait with bigger changes until the PR is approved as desired. Personally I almost never use the default name, so I'd be in favor of this feature, but the proposal itself didn't gain much interest.

dalexeev commented:
See also the [comment](https://github.com/godotengine/godot-proposals/issues/3553#issuecomment-971806667).

akien-mga commented:
Reviewed and approved in PR review meeting, the feature makes sense for users who want to customize this to their own style guides.

Needs a rebase (there's been a PR merged recently which also did style fixes to this code).

dalexeev committed:
Add Default Callback Name editor setting

Adds the "interface/editors/default_signal_callback_name" editor setting,
which allows you to specify the format of the default callback name
in the Signal Connection Dialog.

dalexeev commented:
I think this should be a project setting, not an editor setting, as different projects may have different code style conventions. And in general, we should unify those editor settings that can be overridden in the project settings.

akien-mga reviewed (approved):
It's ready so let's merge, but for future PRs please make sure to keep cosmetic changes separate from functional changes. It's very difficult to review the actual logic changes in this commit in the midst of all the renames.

akien-mga commented:
Thanks!

> I think this should be a project setting, not an editor setting, as different projects may have different code style conventions. And in general, we should unify those editor settings that can be overridden in the project settings.

That's matter for a proposal discussion IMO, and affects a lot more than just this setting (e.g. whether to use tabs or spaces for indentation, etc.).

dalexeev commented:
> It's ready so let's merge, but for future PRs please make sure to keep cosmetic changes separate from functional changes. It's very difficult to review the actual logic changes in this commit in the midst of all the renames.

Okay. Initially I made two commits, but before the rebase I squash them into one (according to the [docs](https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html)).

akien-mga commented:
We often ask to squash commits when they reflect a single semantic change split over multiple check-ins (e.g. you add a method, then in another commit you bind it, then in a third commit you add docs - all this is a single semantic change -- new API -- which should be a single commit). But sometimes we do want to keep multiple commits when they do separate things which are valuable to keep separate, such as splitting a big cosmetic overhaul from actually implementing a feature in its midst.
